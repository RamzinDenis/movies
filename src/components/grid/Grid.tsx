import React, { useEffect } from "react";
import styles from "./grid.module.sass";
import { RootState } from "../../redux/root-reducer";
import { getMoviesList } from "../../redux/movies/movies.selectors";
import { connect } from "react-redux";
import MovieCard from "../movie-card";
import { loadGenres } from "../../redux/genres/genres.sagas";
import { getGenresList } from "../../redux/genres/genres.selectors";
interface GridProps extends ConnectedProps {
	title: string;
}
type ConnectedProps = ReturnType<typeof mapStateToProps> &
	typeof mapDispatchToProps;

const Grid: React.FC<GridProps> = ({ title, movies, loadGenres, genres }) => {
	useEffect(() => {
		loadGenres();
	}, [loadGenres]);
	return (
		<section className={styles.container}>
			<div className={styles.main}>
				<h2 className={styles.title}>{title}</h2>
				<div className={styles.grid}>
					{movies.map(movie => (
						<MovieCard movie={movie} key={movie.id} genresMap={genres} />
					))}
				</div>
			</div>
		</section>
	);
};

const mapStateToProps = (state: RootState) => ({
	movies: getMoviesList(state),
	genres: getGenresList(state),
});

const mapDispatchToProps = {
	loadGenres,
};

export default connect(mapStateToProps, mapDispatchToProps)(Grid);
