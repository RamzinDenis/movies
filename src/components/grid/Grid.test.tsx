import React from "react";
import { shallow } from "enzyme";
import Grid from "./";
import toJson from "enzyme-to-json";

describe("<Grid />", () => {
	const mockMovies = [
		{
			backdrop_path: "string",
			id: 1234,
			title: "Robin Hood",
			overview: "This is the story...",
			vote_average: 7.4,
			release_date: "2019_09-17",
			poster_path: "",
			genre_ids: [12, 14, 10751, 18, 878],
		},
		{
			backdrop_path: "string",
			id: 654637,
			title: "Dostaevskiy",
			overview: "This is the story...",
			vote_average: 7.8,
			release_date: "2020_09-17",
			poster_path: "",
			genre_ids: [12, 14, 10751, 18, 878],
		},
	];
	const gridMockProps = {
		movies: mockMovies,
		title: "Popular movies",
		loadGenres: null as any,
		genres: null as any,
	};
	it("should render properly", () => {
		const component = shallow(<Grid.WrappedComponent {...gridMockProps} />);
		expect(toJson(component)).toMatchSnapshot();
	});
});
