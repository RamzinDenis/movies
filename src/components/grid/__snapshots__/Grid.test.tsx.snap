// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should render properly 1`] = `
<section
  className="container"
>
  <div
    className="main"
  >
    <h2
      className="title"
    >
      Popular movies
    </h2>
    <div
      className="grid"
    >
      <MovieCard
        genresMap={null}
        key="1234"
        movie={
          Object {
            "backdrop_path": "string",
            "genre_ids": Array [
              12,
              14,
              10751,
              18,
              878,
            ],
            "id": 1234,
            "overview": "This is the story...",
            "poster_path": "",
            "release_date": "2019_09-17",
            "title": "Robin Hood",
            "vote_average": 7.4,
          }
        }
      />
      <MovieCard
        genresMap={null}
        key="654637"
        movie={
          Object {
            "backdrop_path": "string",
            "genre_ids": Array [
              12,
              14,
              10751,
              18,
              878,
            ],
            "id": 654637,
            "overview": "This is the story...",
            "poster_path": "",
            "release_date": "2020_09-17",
            "title": "Dostaevskiy",
            "vote_average": 7.8,
          }
        }
      />
    </div>
  </div>
</section>
`;
